DOC: fft.rfft2: add example to docstring

Backfisch263 opened the pull request:
<!-- 
Thanks for contributing a pull request! Please ensure that
your PR satisfies the checklist before submitting:
https://scipy.github.io/devdocs/dev/contributor/development_workflow.html#checklist-before-submitting-a-pr

Also, please name and describe your PR as you would write a
commit message:
https://scipy.github.io/devdocs/dev/contributor/development_workflow.html#writing-the-commit-message.
However, please only include an issue number in the description, not the title,
and please ensure that any code names containing underscores are enclosed in backticks.

Depending on your changes, you can skip CI operations and save time and energy: 
https://scipy.github.io/devdocs/dev/contributor/continuous_integration.html#skipping

Note that we are a team of volunteers; we appreciate your
patience during the review process.

Again, thanks for contributing!
-->

#### Reference issue
https://github.com/scipy/scipy/issues/7168

#### What does this implement/fix?
I have added an example to the `scipy.fft.rfft2` function.

#### Additional information
This is my first pull request on a major repository, any feedback on mistakes is greatly appreciated.

Backfisch263 committed:
Added example to Docstring of scipy.fft.rfft2 function

lucascolley commented on scipy/fft/_basic.py:
```diff
@@ -1166,6 +1166,18 @@ def rfft2(x, s=None, axes=(-2, -1), norm=None, overwrite_x=False, workers=None,
+    --------
+    >>> import scipy.fft
+    >>> import numpy as np
+    >>> x = np.mgrid[:5, :5][0]
```
```suggestion
    >>> x = np.meshgrid(np.arange(5), np.arange(5))[0]
```

lucascolley reviewed (approved):
thanks @Backfisch263, Looks Good To Me!

lucascolley committed:
tweak example [docs only]

lucascolley commented on scipy/fft/_basic.py:
```diff
@@ -1166,6 +1166,18 @@ def rfft2(x, s=None, axes=(-2, -1), norm=None, overwrite_x=False, workers=None,
+    --------
+    >>> import scipy.fft
+    >>> import numpy as np
+    >>> x = np.meshgrid(np.arange(5), np.arange(5))[0]
```
```suggestion
    >>> x = np.meshgrid(np.arange(5), np.arange(5))[1]
```

lucascolley committed:
fix tweak [docs only]

DietBru commented on scipy/fft/_basic.py:
```diff
@@ -1166,6 +1166,18 @@ def rfft2(x, s=None, axes=(-2, -1), norm=None, overwrite_x=False, workers=None,
+    --------
+    >>> import scipy.fft
+    >>> import numpy as np
+    >>> x = np.meshgrid(np.arange(5), np.arange(5))[1]
+    >>> scipy.fft.rfft2(x)
+    array([[ 50.  +0.j        ,   0.  +0.j        ,   0.  +0.j        ],
+           [-12.5+17.20477401j,   0.  +0.j        ,   0.  +0.j        ],
+           [-12.5 +4.0614962j ,   0.  +0.j        ,   0.  +0.j        ],
+           [-12.5 -4.0614962j ,   0.  +0.j        ,   0.  +0.j        ],
+           [-12.5-17.20477401j,   0.  +0.j        ,   0.  +0.j        ]])
```
Do you have any objections to changing the values in a way so integers are generated? E.g.:
```suggestion
    >>> x = np.meshgrid([1, 0, -1, 0], [0, 1, 0, 1])[0]
    >>> scipy.fft.rfft2(x)
    array([[0.+0.j, 8.+0.j, 0.+0.j],
           [0.+0.j, 0.+0.j, 0.+0.j],
           [0.+0.j, 0.+0.j, 0.+0.j],
           [0.+0.j, 0.+0.j, 0.+0.j]])
```

lucascolley committed:
tweak example [docs only]

Co-authored-by: Dietrich Brunn <12721170+DietBru@users.noreply.github.com>

Backfisch263 committed:
adjust example for clarity

Replaced np.meshgrid() with np.broadcast_to() to avoid confusion following the suggestion by j-bowhay

Co-authored-by: Jake Bowhay <60778417+j-bowhay@users.noreply.github.com>

DietBru reviewed (approved):
Looks good. Thank you for your contribution, @Backfisch263!
